### DISCUSSION TITLE
binary_sensor.octopus_energy_electricity_{{METER_SERIAL_NUMBER}}_{{MPAN_NUMBER}}_off_peak

### DISCUSSION THREAD

**Discussion by Leventionz** [BottlecapDave/HomeAssistant-OctopusEnergy#659] [Q&A]
Hi,

What's the difference when using Intelligent Octopus between

`binary_sensor.octopus_energy_electricity_{{METER_SERIAL_NUMBER}}_{{MPAN_NUMBER}}_off_peak` and `binary_sensor.octopus_energy_intelligent_dispatching`

I have noted very infrequently the former is **off** when the latter is **on**

screenshot of `binary_sensor.octopus_energy_electricity_{{METER_SERIAL_NUMBER}}_{{MPAN_NUMBER}}_off_peak` note it is **off** between 09:30-10:00 (highlighted in picture)

![Screenshot 2023-12-27 155552](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/24567397/248e310a-9b36-4731-beea-ded1b841ce5e)

screenshot of `binary_sensor.octopus_energy_intelligent_dispatching` showing it is **on** during the same period

![Screenshot 2023-12-27 155428](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/24567397/d6ae8596-1e44-40d1-8702-fc3e525ed0c7)

As I use this in my automations it is beneficial to know which is a more reliable indicator of off peak periods; to date I had assumed `binary_sensor.octopus_energy_intelligent_dispatching` would only switch to **on** during an off peak period.

Additionally, I can see the Octopus mobile app closely aligns to the `binary_sensor.octopus_energy_intelligent_dispatching`

**Comment by BottlecapDave** [comment]
Hello. This is probably due to the updates of rate and intelligent dispatch information being out of sync. Tbh the off peak sensor should only turn on during the guaranteed off peak time and the dispatching should take into account times granted outside of this period. This is because the data I get from OE only tells half the story as it might be a valid slot by OE, but if your car is not available for charge then you wont get the off peak rate, for which you might want to do additional checks.
